### PR TITLE
Render Tank, Search Players, and fixes

### LINKED ISSUE #96: Tank Rendering For Battle Arena and Armory
**Describe the solution you'd like**
A component that takes the array for tanks and renders them.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### LINKED ISSUE #99: React App Tab Title
**Describe the bug**
The website currently says React App on the 

**To Reproduce**
1. npm start
2. Look at tab name in the top.

**Expected behavior**
Should say FORTUNA.

**Screenshots**
Screenshot (22)

**Additional context**
N/A

### LINKED ISSUE #105: Navbar Imports User's Name and Currency
**Describe the solution you'd like**
Fix the Navbar so that it can correctly pass the user and their currency.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### LINKED ISSUE #106: Search Players searches a list and returns a list
**Describe the solution you'd like**
Build a component that takes a list and compares the user's search keyword to that list. The result will return a new list to display users with that keyword in the name.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### LINKED ISSUE #109: Provide Project Version Number in Fortuna Briefing
**Describe the solution you'd like**
Provide Project Version Number in Fortuna Briefing.

**Describe alternatives you've considered**
N/A

**Additional context**
It's a small thing to do, but I'm planning on doing a workflow bit with this in the near future.

### PR DESCRIPTION
**Pull request by blaiadam**
**Description**
Implements two new components. RenderTank will receive a tank object and render it, once the assets are in place. SearchPlayers component takes all users and returns a list of users according to the user's search term. 

This also fixes the Navbar component so that it can work with our API better. Also, the Fortuna version was put into the briefing section.

There is a commit with ComponentMenu too, but it is unfinished.

Resolves #109 
Resolves #106 
Resolves #105 
Resolves #96 
Resolves #99

**Type of change**
Check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Fix or feature that would cause existing functionality to not work as expected
- [ ] This change requires a update in the design document

**Steps to Verify Functionality**
1. Load up website.
2. Go to the Battle Arena page and press search to view a full list of players.
3. Type into the search bar and click search to refine the list of players.
4. The center of the Battle Arena showcases the tank rendering sample.

**Final Checklist:**
Go down the list and check them off.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the design document (if applicable)
- [x] My changes generate no new warnings
- [x] Pulled updated master branch (if changed since branch creation) and corrected any conflicts, if any occur.

### CODE REVIEW

File: src/armory/Armory.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ class Armory extends React.Component<{||}> {
 				</div>
 				<div className="column armorymiddle">
 					<h1>BIG TANK GUY</h1>
-					<RenderTank tank={ ["Chassis4", "Tread1"] } />
+					<RenderTank tank={ ["moddableLight", "advanceTreads"] } />
```
Comment (Derplime) [inline src/armory/Armory.js:37]: advanceTreads -> **advancedTreads**

File: src/armory/Armory.js [hunk 2]
```diff
@@ -3,22 +3,24 @@
...
+				<Navbar styleName="navbtn" linkName="MainMenu" returnName="Back to Main Menu" pageName="Armory" userName="FRIcker" userCurrency={4233} />
 				<div className="column armoryleft">
 					<h3>Select a Tank to Edit</h3>
 					<select className="dropdownMenu">
-                    				<option defaultValue>Select a Tank</option>
-                    				<option value="Child Consumer">Child Consumer</option>
-                    				<option value="Fast Bang">Fast Bang</option>
-                    				<option value="Biggest Gun">Biggest Gun</option>
-                			</select>
+                    	<option defaultValue>Select a Tank</option>
```
Comment (SecondThread) [inline src/armory/Armory.js:23]: Fix the indentation please

File: src/armory/Armory.js [hunk 3]
```diff
@@ -28,83 +30,91 @@ class Armory extends React.Component<{||}> {
...
 		);
 	}
 }
 
+// Needs to be placed into armoryright
+// <ComponentMenu tank={ ["moddableLight", "machineGun", "machineGun", "shortRangeScanner", "shortRangeJammer", "mine", "c4"]} components={ ["heavy", "laser", "deathRay", "itemScanner", "longRangeJammer"] }/>
+
 export default Armory;
+
+/* Needs to be transferred to ComponentMenu.js
```
Comment (SecondThread) [inline src/armory/Armory.js:53]: Is there a reason for this very large amount of dead code? If not, I think we should get rid of it

File: src/components/ComponentMenu.js [hunk 4]
```diff
@@ -0,0 +1,219 @@
+//@flow strict
+/*
```
Comment (SecondThread) [inline src/components/ComponentMenu.js]: No files of commented out code please :)

File: src/battlearena/BattleArena.js [hunk 5]
```diff
@@ -4,36 +4,45 @@ import * as React from 'react';
 import { Link } from 'react-router-dom';
 import Navbar from '../components/Navbar.js';
 import Leaderboard from '../components/Leaderboard.js';
+import RenderTank from '../components/RenderTank.js';
+import SearchPlayers from '../components/SearchPlayers.js';
+
+// searchName tracks the text the user enters into the searchbar to find players to challenge.
+type State = {|
+	searchName: string
```
Comment (SecondThread) [inline src/battlearena/BattleArena.js:13]: If this is never used, it shouldn't be in the state variable

File: src/battlearena/BattleArena.js [hunk 6]
```diff
@@ -4,36 +4,45 @@ import * as React from 'react';
...
+|};
+
+class BattleArena extends React.Component<State> {
```
Comment (SecondThread) [inline src/battlearena/BattleArena.js:16]: Unless something very confusing is going on here that I'm not aware of, you are actually specifying the type of your Props as the type called 'State'. You need to define the type of your props before the state. See this for docs: https://flow.org/en/docs/react/components/

File: src/components/Leaderboard.js [hunk 7]
```diff
@@ -2,9 +2,9 @@
 
 import * as React from 'react';
 
-type Props = {
+type Props = {|
 	leaderNames: Array<string>;
```
Comment (SecondThread) [inline src/components/Leaderboard.js]: This doesn't really have anything to do with this PR, but the leaderboard should probably not be passed the leaderboard names, it should run an API call to get them and store them in the state instead.

File: src/components/RenderTank.js [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+//
+// Needs to be updated to correctly parse through the assets/art folders.
+
+class RenderTank extends React.Component<Props> {
+
+    constructor(props: Props) {
+        super(props);
+
+        // Tracks category for each component.
+        const componentCategories: Map<string, string> = new Map();
```
Comment (SecondThread) [inline src/components/RenderTank.js]: This is broken, but we talked about this earlier

File: src/components/SearchPlayers.js [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+	// Challenge the player chosen.
+	challengePlayer(): void {
+		console.log("empty");
+	}
+
+	render(): React.Node {
+		return (
+			<div className="searchPlayers">
+				<h6>Click the Player's Name you wish to Challenge</h6>
+				<input type="text" className="inputText" onChange={(e) => this.handleChange(e.target.value)} placeholder="Search Players" />
```
Comment (SecondThread) [inline src/components/SearchPlayers.js:86]: I'm pretty sure you are double-wrapping this and you don't need to be.

Keep this as {(e)=>this.handleChange(...)}

On the next line chance onClick to be {()=>this.handleSearch()}

Then make handleSearch and handleChange actual methods. I.E. handleChange(e): void {...}

File: src/components/SearchPlayers.js [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+
+		this.state = {
+			searchName: "",
+			searchList: [""],
+			displayList: <div className="searchPlayersList"></div>
+		}
+	}
+
+	// When typing in the searchbar, update searchName.
+	handleChange = (e: string) => {
```
Comment (SecondThread) [inline src/components/SearchPlayers.js:36]: I mention how you should change this later, but you need to specify the return type too

**Review by SecondThread — CHANGES_REQUESTED** [review]

**Pull request closed without merging**